app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -261,9 +261,20 @@ class NavigationOverlayFragment : Fragment() {
     }
 
     private fun observeAccountState(): Disposable {
-        return serviceLocator.fxaRepo.accountState.subscribe {
+        val fxaRepo = serviceLocator.fxaRepo
+
+        return fxaRepo.accountState.subscribe {
             when (it) {
-                FxaRepo.AccountState.AUTHENTICATED_WITH_PROFILE -> Unit //TODO
+                FxaRepo.AccountState.AUTHENTICATED_WITH_PROFILE -> {
+                    fxaRepo.accountManager.accountProfile()?.let { profile ->
```

Comment:
If the accountProfile is null, we should log a Sentry error.

Reply:
Good idea, I'll ask Grisha about some of the edge cases with nullability

Reply:
@psymoon On second thought, you can add the non-null Profile into the Account State Observable and never have to deal with the null state.